src/ConnectSdk/Querying/Query.cs

```diff
@@ -50,21 +55,53 @@ public Query(string collection, IEventEndpoint endpoint, AliasedAggregations agg
             GroupBy = groups;
             Interval = interval;
             Timezone = timezone;
-        }
 
-        public virtual Query<TNewResultType> UpdateWith<TNewResultType>(AliasedAggregations aggregations = null, FilteredProperties filters = null, ITimeframe timeframe = null, string[] groups = null, Interval? interval = null, object timezone = null)
+            if (custom == null)
+                return;
+
+            _data = new Dictionary<string, object>();
+
+            foreach (var pair in custom)
+            {
+                _data.Add(pair.Key, pair.Value);
+            }
+        }
+        
+        public virtual Query<TNewResultType> UpdateWith<TNewResultType>(AliasedAggregations aggregations = null, FilteredProperties filters = null, ITimeframe timeframe = null, string[] groups = null, Interval? interval = null, object timezone = null, IEnumerable<KeyValuePair<string, object>> custom = null)
         {
-            return new  Query<TNewResultType>(Collection, Endpoint, aggregations ?? this.Select, filters ?? this.Filter, timeframe ?? this.Timeframe, groups ?? this.GroupBy, interval ?? this.Interval, timezone ?? this.Timezone);
+            return new Query<TNewResultType>(Collection, Endpoint, aggregations ?? this.Select, filters ?? this.Filter, timeframe ?? this.Timeframe, groups ?? this.GroupBy, interval ?? this.Interval, timezone ?? this.Timezone, custom ?? this.Custom);
         }
 
         public virtual async Task<QueryResponse<TResult>> Execute()
         {
              return await Endpoint.Query(Collection, this);
         }
-        
+
+        private void OverrideReservedValue(Dictionary<string, object> dict, string key, object value)
+        {
+            if (value == null)
```

Comment:
If we did go with this - an extension method would be much better.  I'm still not loving it though.